### PR TITLE
Add type definitions for compatibility with typescript projects 

### PR DESCRIPTION
**Pull request by emmclaughlin**

### CODE REVIEW

**Review by harry-wang-faire — COMMENTED** [review]
Looking good! @IanEdington Do you think is important to retain the history of the types too from the origin?

File: package.json [hunk 1]
```diff
@@ -58,5 +60,6 @@
   "dependencies": {
     "color": "^3.1.3",
     "react-reconciler": "^0.26.1"
-  }
+  },
+  "types": "types/index.d.ts"
```
Comment (IanEdington) [inline package.json:64]: Could we move this to the top of this file close to the main / files members?

File: types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,575 @@
+// Type definitions for mjml-react 2.0
+// Project: https://github.com/wix-incubator/mjml-react
+// Definitions by: Henri Normak <https://github.com/henrinormak>
+//                 Ian Edington <https://github.com/IanEdington>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
```
Comment (IanEdington) [inline types/index.d.ts:7]:
```suggestion
```

**Review by IanEdington — APPROVED** [review]
Awesome! LGTM

**Comment by IanEdington** [comment]
> Looking good! @IanEdington Do you think is important to retain the history of the types too from the origin?

Since we'll be replacing this anyways and it's a not core to the logic of the components I don't think the effort is worth it

**Pull request merged**